### PR TITLE
Add common `DockerObject` parent to every model class

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java [hunk 1]
```diff
@@ -53,7 +66,74 @@ default ObjectMapper getObjectMapper() {
...
+    protected JsonDeserializer<?> newDelegatingInstance(JsonDeserializer<?> newDelegatee) {
+        return new DockerObjectDeserializer(newDelegatee, beanDescription, originalMapper);
+    }
+
+    @Override
+    @SuppressWarnings({"deprecation", "unchecked"})
+    public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+        JsonNode jsonNode = p.readValueAsTree();
+
+        Object deserializedObject = originalMapper.treeToValue(jsonNode, beanDescription.getBeanClass());
```
Comment (cowtowncoder) [inline docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java:128]: Jackson 2.13 will  help here with method `DeserializationContext.readTreeAsValue(JsonNode, Class)`, but alas not yet available. But there is `DeserializationContext.readTree(JsonParser)` which has some small benefits over going through `JsonParser` -- probably not much difference here just minor improvement.

Comment (bsideup) [reply docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java:128]: @cowtowncoder thanks for having a look!
I tried `DeserializationContext.readTree(JsonParser)` but ended up getting a recursive problem 😅 Hence the "originalMapper" workaround.

Comment (cowtowncoder) [reply docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java:128]: Ah. Odd... would be interested in details if you have any (just to see if there's something to fix), but makes sense to skip in that case.

Comment (bsideup) [reply docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java:128]: As now I have your aura around, let me try it again and report back :)
Can also share how to try it, without even having Docker running (a.k.a. in memory)

Comment (cowtowncoder) [reply docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java:128]: Sounds good!

Comment (bsideup) [reply docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java:128]: Ah, okay, I am stupid :D I confused `readTree` with `readValue` 😅 

`readValue` gives a recursive problem, while `JsonNode jsonNode = ctxt.readTree(p);` (previous line) works just fine, thanks!

**Pull request merged**